src/awssh/awssh.py

```diff
@@ -0,0 +1,152 @@
+"""awssh simplifies secure shell connections over AWS simple systems manager.
+
+EXIT STATUS
+    This utility exits with the same status as the underlying aws or ssh process:
+    0   No error.
+    >0  An error occurred.
+
+Usage:
+  awssh [options] <profile> <instance-id> [<command>...]
+  awssh (-h | --help)
+
+Options:
+  -c --credentials=FILENAME     Shared credentials file name.
+  -n --no-ssh                   Open an SSM shell session without using ssh.
+  -r --region=REGION            AWS region name to use.
+  -s --ssh-args=ARGUMENTS       Arguments to send to ssh.
+  --log-level=LEVEL             If specified, then the log level will be set to
+                                the specified value.  Valid values are "debug", "info",
+                                "warning", "error", and "critical". [default: info]
+  -h --help                     Show this message.
+"""
+
+
+# Standard Python Libraries
+import logging
+import os
+from pathlib import Path
+import subprocess  # nosec: B404 subprocess use is required for this tool
+import sys
+from typing import Any, Dict, Optional
+
+# Third-Party Libraries
+import docopt
+
+# I cannot find a package that includes type stubs for schema, so I must add
+# "type: ignore" to tell mypy to ignore this library
+from schema import And, Schema, SchemaError, Use  # type: ignore
```

Comment:
I'd prefer to leave out `ignore`s that are only necessary when running `mypy` manually.  I never run `mypy` manually, so such a requirement would mean I would have to do so in order to figure out what breaks.

OTOH, I'd be OK with requiring these `ignore`s if we dropped the `--ignore-missing-imports` option from the `pre-commit` hook.  That way they would be noticed without having to rely on my abysmal memory to perform a manual step (i.e., running `mypy` manually).

Reply:
I'm fine with updating the default arguments for the `mypy` hook (`["--ignore-missing-imports", "--scripts-are-modules"]`) so that it fails on imports. It would ensure that we are performing the best `mypy` analysis possible. I typically run `mypy` manually only to determine the stubs necessary to add to the pre-commit configuration, but otherwise only rarely run it manually (mostly for type discovery with `reveal_type()` and `reveal_locals()`).

Reply:
The reason I added the `ignore` is because I use `setup-env` to create my environment, and [it runs `mypy`](https://github.com/cisagov/awssh/blob/first-commits/setup-env#L157-L158) (without `--ignore-missing-imports`) and outputs these errors:
```
src/awssh/autocompleter.py:13: error: Skipping analyzing "boto3": module is installed, but missing library stubs or py.typed marker
src/awssh/autocompleter.py:13: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/awssh/awssh.py:33: error: Skipping analyzing "docopt": module is installed, but missing library stubs or py.typed marker
src/awssh/awssh.py:34: error: Skipping analyzing "schema": module is installed, but missing library stubs or py.typed marker
```

If we are providing a setup script for people to use, it shouldn't output any error messages, even if they are just from `mypy`.

Reply:
@mcdonnnj - Where do we configure the mypy pre-commit hook to `--ignore-missing-imports`?  I'm not seeing that.

Reply:
@dav3r That's one of the options it's run with by default from the hook [here](https://github.com/pre-commit/mirrors-mypy/blob/730f5f57bd6fae58e10e6d8d421b945e88a62fed/.pre-commit-hooks.yaml#L7).

Reply:
@mcdonnnj and @jsf9k - are we all on board with getting rid of `--ignore-missing-imports` from the mypy pre-commit hook and adding `ignore` hints where necessary?

Reply:
Note that that is really an upstream change.

Reply:
Understood, though I'd like to make it locally in this PR just to get it taken care of here.

Reply:
This has been taken care of in https://github.com/cisagov/awssh/pull/9/commits/2467d06efcee16531c0c80529e6c02e4cedb7f7b.

https://github.com/cisagov/skeleton-generic/issues/103 has been created to handle this upstream.